testsuite/classlibrary/TestEnv.sc

```diff
@@ -26,6 +35,85 @@ TestEnv : UnitTest {
 			"Exponential Env.cutoff should be same as linear Env.cutoff.linexp");
 	}
 
+	test_noff_bug334 {
+		var synth, n_off_resp, n_end_resp, passed,
+		cond = Condition.new,
+		cleanup = {
+			n_off_resp.free;
+			n_end_resp.free;
+			server.quit;
+		};
+		this.bootServer(server);
+		synth = {
+			// doneAction: 1 should reach end *after* doneAction: 2 envelope ends
+			// correct behavior is that '/n_off' is never sent
+			EnvGen.kr(Env([0, 1], [0.2]), gate: 0, doneAction: 1);
+			EnvGen.kr(Env([0, 1], [0.1]), doneAction: 2);
+		}.play(server);
+		n_off_resp = OSCFunc({
+			cleanup.value;
+			passed = false;
+			cond.unhang;
+		}, '/n_off', server.addr, argTemplate: [synth.nodeID]);
+		// responsible for cleaning up this test
+		// n_off_resp should NOT fire! so we need another for the stop condition
+		n_end_resp = OSCFunc({
+			cleanup.value;
+			passed = true;
+			cond.unhang;
+		}, '/n_end', server.addr, argTemplate: [synth.nodeID]);
+		cond.hang;
+		this.assert(passed, "EnvGen: Initial gate == 0 with doneAction == 1 should not send /n_off");
+	}
+
+	// has failed in supernova
+	// methodology: synth should play for 0.1 sec if it's working
+	// so we will send a trigger just before the envelope ends
+	// if the bug occurs, the synth will end almost immediately -- no trigger
+	test_nongated_bug3094 {
```

Comment:
`test_EnvGen_impulse_gate_shouldnt_close_nongated_envelope`

Someone else might have a better idea about wording, but my general suggestion is following review comments I've gotten on other unit tests.